test/built-ins/Temporal/Calendar/fields.js

```diff
@@ -0,0 +1,27 @@
+// Copyright (C) 2021 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.calendar.prototype.fields
+description: Temporal.Calendar.prototype.fields will take iterable of any size and any string
+  and return Array of the same content.
+info: |
+  ## 12.4.21 Temporal.Calendar.prototype.fields ( fields )
+  1. Let calendar be the this value.
+  2. Perform ? RequireInternalSlot(calendar, [[InitializedTemporalCalendar]]).
+  3. Assert: calendar.[[Identifier]] is "iso8601".
+  4. Let fieldNames be ? IterableToListOfType(fields, « String »).
+  5. Return ! CreateArrayFromList(fieldNames).
+features: [Temporal]
+includes: [deepEqual.js]
+---*/
+let cal = new Temporal.Calendar("iso8601")
+const fields = {
+  *[Symbol.iterator]() {
+      let i = 0;
+      while (i++ < 1000001) {
+        yield "garbage " + i;
+      }
+  }
+}
+assert.deepEqual(cal.fields(fields), Array.from(fields));
```

Comment:
It doesn't seem like `deepEqual` actually checks the result is an Array. Not sure why we'd use it over `compareArray`

Reply:
DONE